crates/sdk/src/prover/vm/local.rs

```diff
@@ -36,26 +37,46 @@ impl<SC: StarkGenericConfig, VC, E: StarkFriEngine<SC>> VmLocalProver<SC, VC, E>
         Self {
             pk,
             committed_exe,
-            overridden_heights: None,
+            continuation_heights: None,
+            single_segment_heights: None,
             _marker: PhantomData,
         }
     }
 
-    pub fn new_with_overridden_trace_heights(
+    pub fn new_with_overridden_continuation_trace_heights(
         pk: Arc<VmProvingKey<SC, VC>>,
         committed_exe: Arc<VmCommittedExe<SC>>,
         overridden_heights: Option<VmComplexTraceHeights>,
     ) -> Self {
         Self {
             pk,
             committed_exe,
-            overridden_heights,
+            continuation_heights: overridden_heights,
+            single_segment_heights: None,
             _marker: PhantomData,
         }
     }
 
-    pub fn set_override_trace_heights(&mut self, overridden_heights: VmComplexTraceHeights) {
-        self.overridden_heights = Some(overridden_heights);
+    pub fn new_with_overridden_single_segment_trace_heights(
```

Comment:
maybe we should just have a `default()` and follow the builder pattern which consumes `mut self` and returns `Self` to do `with_overridden_single_segment_trace_heights()` and `with_single_segment_trace_heights()`